src/main/java/org/jvnet/hudson/test/RestartableJenkinsRule.java

```diff
@@ -188,9 +208,20 @@ public File allocate() throws Exception {
         };
 
         // run each step inside its own JenkinsRule
-        for (Statement step : steps) {
+        for (Map.Entry<Statement, Boolean> entry : steps.entrySet()) {
+            Statement step = entry.getKey();
             j = createJenkinsRule(description).with(loader);
-            j.apply(step,description).evaluate();
+            try {
+                j.apply(step, description).evaluate();
+                if (!entry.getValue()) {
+                    Assert.fail("The current JenkinsRule should have failed to start Jenkins.");
```

Comment:
Super-🐜 - I like to phrase this as "X happened but we saw Y" because that makes it clearer.